### PR TITLE
Item/Recipe Additions, Changes

### PR DESCRIPTION
**Pull request by chaosvolt**
A couple things thought up or observed.

Item Additions:
1. Flatbread. Simpler and less filling alternative to normal leavened bread.
2. A greatcoat. Since we have peacoats instead of wool trenchcoats, one could regard this as the duster equivalent.
3. Windbreaker jacket. Unlike the evac jacket, it's VAR_SIZE and also rainproof. But also less warmth.

Recipe tweaks:
1. Added flatbread to cooking recipes that use bread, cornbread, and other options.
2. Added an option to dismantle spikes, scrap suits, springs, wires, and blades.

### CODE REVIEW

File: data/json/items/armor.json [hunk 1]
```diff
@@ -4139,6 +4139,30 @@
...
+        "storage" : 2,
+        "symbol" : "[",
+        "description" : "A light synthetic jacket with a hood.  Not very warm, but will keep out the rain.",
+        "price" : 1000,
+        "material" : ["plastic"],
+        "volume" : 9,
+        "cutting" : 0,
+        "warmth" : 10,
+        "phase" : "solid",
+        "environmental_protection" : 0,
```
Comment (Coolthulhu) [inline data/json/items/armor.json]: I'd buff it to 1 or 2, because it's made of plastic and waterproof.

File: data/json/items/comestibles.json [hunk 2]
```diff
@@ -11622,6 +11622,35 @@
...
+        "spoils_in" : 280,
+        "stim" : 0,
+        "container" : "bag_plastic",
+        "to_hit" : 0,
+        "comestible_type" : "FOOD",
+        "symbol" : "%",
+        "quench" : 0,
+        "healthy" : 1,
+        "addiction_potential" : 0,
+        "nutrition" : 15,
```
Comment (Coolthulhu) [inline data/json/items/comestibles.json]: 3 flours for 8*15=120 nutrition. How does it compare to similar food products, such as johnnycake?

That healthy 1 doesn't fit, fried flour with water isn't exactly healthy.

File: data/json/items/comestibles.json [hunk 3]
```diff
@@ -11622,6 +11622,35 @@
...
+        "price" : 140,
+        "material" : "wheat",
+        "tool" : "null",
+        "volume" : 2,
+        "cutting" : 0,
+        "phase" : "solid",
+        "charges" : 8,
+        "bashing" : 0,
+        "flags" : ["EATEN_HOT"],
+        "fun" : 1
```
Comment (Coolthulhu) [inline data/json/items/comestibles.json]: I can't really think of food blander than fried flour... 0 would fit better. Also that "EATEN_HOT" doesn't fit.

**Comment by Barhandar** [comment]
> 1. Made recipe for spike and scrap suit reversible.
> 2. Added an option to dismantle springs, wires, and blades.

You mean, disabled batch-crafting of spikes, springs, wires and blades?

**Comment by BevapDin** [comment]
>> Made recipe for spike and scrap suit reversible.
>> Added an option to dismantle springs, wires, and blades.

> You mean, disabled batch-crafting of spikes, springs, wires and blades?

It only disables batch-crafting for spikes and scrap suit. Springs, wires, and blades have separate uncraft recipes, so they are technically not reversible (in the sense of directly reverting what has been crafted). The uncraft recipe can yield items completely unrelated to the crafting recipe.

**Comment by Barhandar** [comment]
>It only disables batch-crafting for spikes

Which, last time I checked, are vehicle-installable and might be used in quantity. So I still see no point in making them un-batchable.

Speaking of which, why is batch-crafting disabled for reversible items, in the first place? Batch only reduces time requirement, not tool charges nor ingredients. And disablement makes crafting stuff which is reversible for no reason (looking at you, sandwiches) annoying.

**Comment by Coolthulhu** [comment]
> Speaking of which, why is batch-crafting disabled for reversible items, in the first place? Batch only reduces time requirement, not tool charges nor ingredients.

I don't think it is like that for any balance-related reasons. Probably a technical thing. I'll check if it's easy to change, because it looks like a big "feature placeholder".

**Comment by chaosvolt** [comment]
Ack. I could add separate uncraft recipes for spikes and scrap suits then, just to underp that. >.>

As for flatbread, note that regular bread has 18 nutrition, which is the main thing I was comparing it to.

**Comment by Barhandar** [comment]
>As for flatbread, note that regular bread has 18 nutrition, which is the main thing I was comparing it to.

_Corn_bread has 18 nutrition (and also 4 portions instead of 8). Regular "bread" has 20.

**Comment by chaosvolt** [comment]
...doh. Ah right, read that wrong.

It seems the general pattern is that flour-based breads and similar have a stack size of 8, and cord-based ones have a stack size of 4.

**Comment by chaosvolt** [comment]
...yep, I'm clearly up way too early. Reducing flatbread's fun and healthiness to zero is a nerf, not a buff. >.>

**Comment by Barhandar** [comment]
>Added flatbread to recipes that allow the use of bread and/or
alternatives.

Can lua scripts be used in mainline data, outside of mods?

Can lua scripts modify recipe components?

**Comment by chaosvolt** [comment]
Wat. This has a negative amount of things to do with an item addition in mainline content. :V

**Comment by Coolthulhu** [comment]
I don't like the idea of flatbread sandwiches with same stats as regular bread ones for lower cost.

If you want flatbread sandwiches, change it so that flatbread piece has the same nutrition as regular one, but less charges for same amount of flour.

**Comment by Barhandar** [comment]
>Wat. This has a negative amount of things to do with an item addition in mainline content. :V

It has to do with modifying one place with recipe instead of a dozen every time a certain kind of item is added. Since distributed handling - #11051, which allows setting a parameter in the item's definition without modifying recipes at all - is unlikely to get implemented anytime soon, I'd settle for being able to modify a Lua script, which then adjusts recipes when they're loaded.
>I don't like the idea of flatbread sandwiches with same stats as regular bread ones for lower cost.

Wastebread. You're welcome!

**Comment by chaosvolt** [comment]
Wastebread does add a touch of lunacy to everything, but yeah. I guess I could lower charges instead of nutirition.

Though every option for sandwiches has different nutrition. Should we make different recipes for hardtack too? >.>

**Comment by Barhandar** [comment]
>I guess I could lower charges instead of nutirition.

Or do the same thing hardtack does, make it require twice the ingredients for the craft.

**Comment by chaosvolt** [comment]
...for flatbread? It's tortillas, basically. >.>

**Comment by Coolthulhu** [comment]
> Wastebread. You're welcome!

Checked it out.
Wastebread is now declared a balance crime.

**Comment by chaosvolt** [comment]
Wastebread. Guaranteed to add a touch of absurdity to any discussion.

In any case, I guess I'll go with 20 nutrition and a stack of 6. Same net nutrition per stack (which is still less than a stack of bread) without causing sandwich lunacy like wastebread does.

**Comment by Barhandar** [comment]
>...for flatbread? It's tortillas, basically. >.>

So is hardtack, which is, by the way, preparable without yeast and only uses 1.5x more flour for the same amount of resulting floor than plain bread. And it needs salt, which only requires a single visit to the swamps (and a lot of time, but...).  
So basically the flatbread, here, is hardtack which had salt exchanged for the flour. And made to spoil. Not exactly a favorable comparsion as long as salt is renewable.
> Wastebread is now declared a balance crime.

It would be if it didn't have more complex preparation than "find some flour-resulting seeds, plant them, get tons of flour after a season" of other breads.

**Comment by chaosvolt** [comment]
Hardtack also takes stupidly long to prepare. Not really sure how to balance flatbread relative to hardtack, but balancing it regular to normal bread is infinitely simpler. >.>

**Comment by Barhandar** [comment]
>Hardtack also takes stupidly long to prepare. 

Once. It's imperishable, so you can do it one time and then not worry about having bread for sandwiches for years, and not get annoyed by needing to sequentally craft things without a (semi-auto) crafting queue.

**Comment by chaosvolt** [comment]
I've found it only practical to make them 5 at a time, actually. 30 hours to make 20 stacks, last I checked.

**Comment by Barhandar** [comment]
>I've found it only practical to make them 5 at a time, actually. 30 hours to make 20 stacks, last I checked.

Sounds like it needs a dose of batch-time-factor. After all, it should be possible to do several pieces of bread at once.

**Comment by chaosvolt** [comment]
Would be nice, yes. Oh derp, I'm doing recipes tweaks and additions, I could be doing that. >_>

**Comment by chaosvolt** [comment]
Ahah, finally found that biscuits have a batch time factor. Was gonna be reluctant to give bread-type foods batching ability if none had that property.

**Comment by chaosvolt** [comment]
Ah, hope it goes well.

**Comment by Coolthulhu** [comment]
So, hoodies. Not mergeable until they both act the same way.
I'd drop the change for now and maybe later retry in a separate PR.

**Comment by chaosvolt** [comment]
Ah, okay. Guess I'll drop the change then, yeah. Should I now, wait for you to de-assign, or let you mess with it?

**Comment by Coolthulhu** [comment]
If you're still on the branch, push it now.

**Comment by chaosvolt** [comment]
Ah, okay.

**Comment by Coolthulhu** [comment]
By the way, why is a spike/suit recipe a separate one rather than just adding "reversible": true to the original one?

**Comment by chaosvolt** [comment]
Knee-jerk reaction on my part. Unlike spikes, I would assume disabling batch crafting for scrapsuits wouldn't be as annoying? @_@

**Comment by Coolthulhu** [comment]
BevapDin already fixed that and I merged the fix. It no longer disables batch crafting.

**Comment by chaosvolt** [comment]
Doh. Since spikes use the same hammering quality needed to make them, I'm guessing even that can be changed to a reversible recipe.

Blades, wires, and springs use different tools to make though, so their uncraft recipes probably should stay.

Guess I'll fix that then?

**Comment by Coolthulhu** [comment]
Sounds about right.
What about scrap suit?

**Comment by chaosvolt** [comment]
Scrap suit can be made reversible, yeah. Even before that batch thing was fixed, it wasn't really something you'd be likely t o craft and then uncraft in bulk.

**Comment by Coolthulhu** [comment]
Merge now? Wait for spike/scrap reversal fix?

**Comment by chaosvolt** [comment]
Is syncing now.

**Comment by chaosvolt** [comment]
There we go. Sorry about that. x.x

**Comment by Coolthulhu** [comment]
Minor: barbed wire now stands out as non-uncraftable

**Comment by chaosvolt** [comment]
Ack, back. And...huh, that's an oddity. Didn't even know it was uncraftable.

And thank you for the merge. ^^"

**Pull request merged**